actions/generate-schedule/main.go

```
@@ -0,0 +1,126 @@
+
+	// One of these flags is required
+	start                = flag.String("start", "", "Start generating from this date. If this flag and previousSchedule are not specified, starts from tomorrow's date.")
+	previousSchedulePath = flag.String("previousSchedule", "", "Path to a previous schedule's YAML file.")
+
+	// TODO(ttomsu): Flag is required now, make optional later (once GH teams integration works).
+	userList = flag.String("users", "", "Comma-separated list of users in the rotation. Usernames will be alphabetically sorted prior to assigning shifts.")
+
+	outFile = flag.String("out", "", "File to write out new schedule. Will overwrite the file if it exists.")
+)
+
+const (
+	startStopFormat = "2006-01-02"
+	usersSeparator  = ","
+)
+
+func main() {
+	flag.Parse()
+
+	if err := validateFlags(); err != nil {
+		log.Fatalf("Error validating flags: %v", err)
+	}
+
+	userSlice := strings.Split(*userList, usersSeparator)
+	userSource := users.NewStaticSource(userSlice...)
+
+	sched, err := scheduler.NewScheduler(userSource, *shiftDurationDays)
+	if err != nil {
+		log.Fatalf("Error creating scheduler: %v", err)
+	}
+
+	stopTime, _ := time.Parse(startStopFormat, *stop)
+
+	var newSchedule *schedule.Schedule
+	if *previousSchedulePath != "" {
+		previousSchedule, err := parsePreviousSchedule(*previousSchedulePath)
+		if err != nil {
+			log.Fatalf("Error parsing previous schedule: %v", err)
+		}
+
+		newSchedule, err = sched.ExtendSchedule(previousSchedule, stopTime)
+		if err != nil {
+			log.Fatalf("Error extending schedule: %v", err)
+		}
+	} else {
+		var startTime time.Time
+		if *start != "" {
+			startTime, _ = time.Parse(startStopFormat, *start)
+		} else {
+			startTime = time.Now().AddDate(0, 0, 1) // tomorrow.
+		}
+
+		newSchedule, err = sched.Schedule(startTime, stopTime)
+		if err != nil {
+			log.Fatalf("Error generating new schedule: %v", err)
+		}
+	}
+
+	scheduleBytes, err := newSchedule.MarshalYAML()
```

Looks like we're ignoring this error and still trying to write the file in the next line.

---

Oops - you're right. Fixed.